### PR TITLE
RavenDB-10269 Make sure we wait until callbacks have completed when d…

### PR DESCRIPTION
**Pull request by arekpalinski**
…isposing the timer

### CODE REVIEW

File: src/Sparrow/Utils/NativeMemoryCleaner.cs [hunk 1]
```diff
@@ -81,7 +80,18 @@ public void CleanNativeMemory(object state)
...
+#if !NETSTANDARD1_3
+            using (var waitHandle = new ManualResetEvent(false))
+            {
+                if (_timer.Dispose(waitHandle))
+                {
+                    waitHandle.WaitOne();
+                }
+            }
+#else
+            Monitor.Enter(_lock); // prevent from running the callback _after_ dispose
```
Comment (ayende) [inline src/Sparrow/Utils/NativeMemoryCleaner.cs:92]: Not a good idea. If you are running into this, it will hang the timer thread.

Comment (arekpalinski) [reply src/Sparrow/Utils/NativeMemoryCleaner.cs:92]: The timer thread (in the callback method) just calls:
`Monitor.TryEnter(_lock, ref lockTaken);` 

https://github.com/ravendb/ravendb/blob/789e53205e9d1c46e725dd60da02b84bc955f3cd/src/Sparrow/Utils/NativeMemoryCleaner.cs#L28-L30

So it won't hang it.

Comment (ayende) [reply src/Sparrow/Utils/NativeMemoryCleaner.cs:92]: what happens if Dispose is called twice?

Comment (arekpalinski) [reply src/Sparrow/Utils/NativeMemoryCleaner.cs:92]: Shoudn't be the case but it's error prone now. I saw you fixed it. Thanks.

**Pull request merged**